internal/pkg/term/syncbuffer/syncbuffer.go

```diff
@@ -0,0 +1,99 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package syncbuffer provides a goroutine safe bytes.Buffer as well printing functionality to the terminal.
+package syncbuffer
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+)
+
+// SyncBuffer is a synchronized buffer that can be used to store output data and coordinate between multiple goroutines.
+type SyncBuffer struct {
+	bufMu sync.Mutex    // bufMu is a mutex that can be used to protect access to the buffer.
+	buf   bytes.Buffer  // buf is the buffer that stores the data.
+	done  chan struct{} // done is a channel that can be used to signal when the operations are complete.
+}
+
+// New creates and returns a new SyncBuffer object with an initialized 'done' channel.
+func New() *SyncBuffer {
+	return &SyncBuffer{
+		done: make(chan struct{}),
+	}
+}
+
+// Write appends the given bytes to the buffer.
+func (b *SyncBuffer) Write(p []byte) (n int, err error) {
+	b.bufMu.Lock()
+	defer b.bufMu.Unlock()
+	return b.buf.Write(p)
+}
+
+// IsDone returns true if the Done channel has been closed, otherwise return false.
+func (b *SyncBuffer) IsDone() bool {
+	select {
+	case <-b.done:
+		return true
+	default:
+		return false
+	}
+}
+
+// MarkDone closes the Done channel.
+func (b *SyncBuffer) MarkDone() {
+	close(b.done)
+}
+
+// LabeledSyncBuffer is a struct that combines a SyncBuffer with a string label.
+type LabeledSyncBuffer struct {
+	label   string
+	syncBuf *SyncBuffer
+}
```

Comment:
nit: consider embedding `*SyncBuffer` instead since a `LabeledSyncBuffer` **_is a_** `SyncBuffer`:
```suggestion
type LabeledSyncBuffer struct {
	label   string
	*SyncBuffer
}
```
this way for example in the `Print` method of the `ltp` you can write:
```go
logs := buf.lines() // instead of logs := buf.syncBuf.lines()
```